satpy/writers/cf_writer.py

```diff
@@ -822,6 +822,17 @@ def save_datasets(self, datasets, filename=None, groups=None, header_attrs=None,
                 header_attrs = flatten_dict(header_attrs)
             root.attrs = encode_attrs_nc(header_attrs)
 
+        # Add header attributes needed for satpy_cf_reader
+        root.attrs["instrument"] = datasets[0].attrs["sensor"]
```

Comment:
This `instrument` thing should be fixed by #2176. Could you remove it from this PR and we'll try a merge of 2176 and remerge master into this PR.